plugin/clang/utils.py

```diff
@@ -420,34 +423,39 @@ def build_objc_message_info_details(cursor):
         return result
 
     @staticmethod
-    def cleanup_comment(raw_comment):
+    def cleanup_comment(raw_comment, brief_comment_to_remove):
         """Cleanup raw doxygen comment."""
-        lines = raw_comment.split('\n')
-        clean_lines = []
-        prev_line = ''
-        is_brief_comment = False
-        non_brief_found = False
-        for line in lines:
-            clean = line.strip()
-            if clean.startswith('/'):
-                clean = clean[3:]
-            elif clean.startswith('*'):
-                clean = clean[2:]
-            prev_line = clean
-            if clean[1:].startswith('brief'):
-                is_brief_comment = True
-                continue
-            if clean == '':
-                is_brief_comment = False
-            if clean == '' and prev_line == '':
-                # We want to ignore trailing empty lines.
-                continue
-            if is_brief_comment:
-                continue
-            if not non_brief_found:
-                non_brief_found = True
-                is_brief_comment = True
-                continue
-            non_brief_found = True
-            clean_lines.append(clean)
+        def find_brief_comment(comment_lines, brief_comment_to_remove):
+            for idx, line in enumerate(comment_lines):
+                if line.endswith(brief_comment_to_remove):
+                    return idx
+            raise ValueError("comment_lines does contain a brief comment")
```

Comment:
While trying to solve this problem, I discovered that `cursor.brief_comment` can contain multiple lines from the raw comment joined into one line. The current logic assumes that a brief comment can not span multiple lines, thus causing the exception.

I'll have to rethink the way im locating and removing the brief comment within the raw comment. Another option is to let aside for the separation between brief and raw, and just show the full comment altogether. What do you think?